binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java

```diff
@@ -64,8 +64,13 @@ public void initialize() {
         Configuration config = getThing().getConfiguration();
 
         location = (String) config.get("location");
-        if ("us".equalsIgnoreCase((String) config.get("unit"))) {
+        if ("metric".equalsIgnoreCase((String) config.get("unit"))) {
+            unit = "c";
+        } else if ("us".equalsIgnoreCase((String) config.get("unit"))) {
             unit = "f";
+        } else {
+            // means unit not known
+            unit = null;
```

Comment:
a) yes, once we have the new "MIS/NOT-CONFIGURED" status, we should use it here.
b) wrong. This line in the code deals with the fact that the config property is invalid for some reason. So the default is NOT applied by the framework and the code deliberately sets the unit to "null", which results in requests that return fahrenheit values.